src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -0,0 +1,231 @@
+    }
+
+    /**
+     * Process bulk action for categories status enabling.
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processBulkStatusEnableAction(Request $request)
+    {
+        $this->updateBulkStatus($request->request->get('categories_bulk'), CategoryStatus::ENABLED);
+
+        return $this->redirectToRoute('admin_category_listing');
+    }
+
+    /**
+     * Process bulk action for categories status disabling.
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processBulkStatusDisableAction(Request $request)
+    {
+        $this->updateBulkStatus($request->request->get('categories_bulk'), CategoryStatus::DISABLED);
+
+        return $this->redirectToRoute('admin_category_listing');
+    }
+
+    /**
+     * Show category for editing.
+     *
+     * @param int $categoryId
+     *
+     * @return Response
+     */
+    public function editAction($categoryId)
+    {
+        return $this->redirect(
+            $this->getAdminLink('AdminCategories', [
+                'id_category' => $categoryId,
+                'updatecategory' => 1,
+            ])
+        );
+    }
+
+    /**
+     * Update categories status.
+     *
+     * @param int[] $categoryIds
+     * @param string $newStatus
+     */
+    protected function updateBulkStatus(array $categoryIds, $newStatus)
+    {
+        try {
+            $command = new UpdateCategoriesStatusCommand(
+                $categoryIds,
+                new CategoryStatus($newStatus)
+            );
+
+            $this->getCommandBus()->handle($command);
+
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (CategoryException $e) {
+            $this->addFlash('error', $this->handleUpdateStatusException($e));
+        }
+    }
+
+    /**
+     * Handle exception when which occurs when updating category status.
+     *
+     * @param CategoryException $e
+     *
+     * @return string Error message
+     */
+    protected function handleUpdateStatusException(CategoryException $e)
+    {
+        $type = get_class($e);
+
+        $errors = [
+            CategoryNotFoundException::class => sprintf(
+                '%s %s',
+                $this->trans('An error occurred while updating the status for an object.', 'Admin.Notifications.Error'),
+                $this->trans('(cannot load object)', 'Admin.Notifications.Error')
+            ),
+            CannotUpdateCategoryStatusException::class => $this->trans('An error occurred while updating the status for an object.', 'Admin.Notifications.Error'),
+        ];
+
+        if (isset($errors[$type])) {
+            return $errors[$type];
+        }
+
+        return $this->trans('Failed to update the status', 'Admin.Notifications.Success');
```

Comment:
_Admin.Notifications.Success_? Since it is a notification about a failure, I think it should be _Admin.Notifications.Error_. :-)

Reply:
@LouiseBonnard displaying an error message here is already a success (for us) hihihi